packages/contracts-bedrock/contracts/L1/SystemConfig.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {
+    OwnableUpgradeable
+} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { Semver } from "../universal/Semver.sol";
+
+/**
+ * @title SystemConfig
+ * @notice This contract is used to update L2 configuration via L1
+ */
+contract SystemConfig is OwnableUpgradeable, Semver {
+    uint256 public constant VERSION = 0;
+
+    uint256 public overhead;
+    uint256 public scalar;
+    address public batcher;
+
+    event ConfigUpdate(
+        uint256 indexed version,
+        UpdateType indexed updateType,
+        bytes data
+    );
+
+    enum UpdateType {
+        BATCHER,
+        GAS_CONFIG
+    }
+
+    constructor(uint256 _overhead, uint256 _scalar, address _batcher) Semver(0, 0, 1) {
+        overhead = _overhead;
+        scalar = _scalar;
+        batcher = _batcher;
+    }
+
+    /**
```

Comment:
Would you consider this to be the canonical way we do upgradable ownable contracts?
cc @maurelian @smartcontracts

Reply:
In slack there was some soft agreement that following the l2 output oracle pattern works for now. We can revisit it in a separate PR.